python/examples/integrations/kafka-example/Kafka_Example.ipynb

```diff
@@ -0,0 +1,832 @@
+    "warnings.simplefilter(\"ignore\")\n",
+    "\n",
+    "producer = KafkaProducer(bootstrap_servers='localhost:9092',\n",
+    "                         value_serializer=lambda v: json.dumps(v).encode('utf-8'))\n",
+    "\n",
+    "\n",
+    "data_file = \"lending_club_demo.csv\"\n",
+    "full_data = pd.read_csv(os.path.join(data_file))\n",
+    "\n",
+    "data = full_data[full_data['issue_d'] == 'Jan-2017']\n",
+    "\n",
+    "for i, row in data.iterrows():\n",
+    "    producer.send('whylogs-stream', row.to_dict())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Consuming the messages with KafkaConsumer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 177,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "topic whylogs-stream - partition 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "from kafka import KafkaConsumer, TopicPartition\n",
+    "\n",
+    "\n",
+    "consumer = KafkaConsumer(bootstrap_servers='localhost:9092', \n",
+    "                         value_deserializer=lambda x: json.loads(x.decode('utf-8')))\n",
+    "\n",
+    "assignments = []\n",
+    "topics=['whylogs-stream']\n",
+    "\n",
+    "for topic in topics:\n",
+    "    partitions = consumer.partitions_for_topic(topic)\n",
+    "    for p in partitions:\n",
+    "        print(f'topic {topic} - partition {p}')\n",
+    "        assignments.append(TopicPartition(topic, p))\n",
+    "consumer.assign(assignments)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Profiling with `whylogs`\n",
+    "\n",
+    "For the sake of simplicity, we will build a `pandas.DataFrame` from the read messages and then profile and merge until there aren't more messages in the topic."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 178,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TopicPartition(topic='whylogs-stream', partition=0) - 100\n",
+      "TopicPartition(topic='whylogs-stream', partition=0) - 100\n",
+      "TopicPartition(topic='whylogs-stream', partition=0) - 68\n",
+      "TopicPartition(topic='whylogs-stream', partition=0) - 41\n",
+      "total 309\n"
+     ]
+    }
+   ],
+   "source": [
+    "import whylogs as why\n",
+    "import pandas as pd \n",
+    "\n",
+    "\n",
+    "consumer.seek_to_beginning()\n",
+    "\n",
+    "total = 0 \n",
+    "counter = 0\n",
+    "while True:\n",
+    "    finished = True\n",
+    "    record = consumer.poll(timeout_ms=500, max_records=100, update_offsets=True)\n",
+    "    for k,v in record.items():\n",
+    "        print(f'{k} - {len(v)}')\n",
+    "        df = pd.DataFrame([row.value for row in v])\n",
+    "        if counter == 0:\n",
+    "            profile = why.log(df).profile()\n",
+    "        else:\n",
+    "            profile.track(df)\n",
```

Comment:
Can we configure a rolling logger here rather than guiding users to manage the profile themselves? Ideally users don't have to know about `profile.track()`

Reply:
see if that is better now @jamie256